packages/zowe-explorer-api/src/vscode/ZoweVsCodeExtension.ts

```diff
@@ -180,11 +184,9 @@ export class ZoweVsCodeExtension {
             // If base profile already has a token type stored, then we check whether or not the connection details are the same
             (serviceProfile.profile.host === baseProfile.profile.host && serviceProfile.profile.port === baseProfile.profile.port);
         // If the connection details do not match, then we MUST forcefully store the token in the service profile
-        let profileToUpdate: imperative.IProfileLoaded;
+        let profileToUpdate = serviceProfile;
         if (connOk) {
-            profileToUpdate = baseProfile;
-        } else {
-            profileToUpdate = serviceProfile;
+            profileToUpdate = serviceProfile.name.startsWith(baseProfile.name + ".") ? { ...baseProfile, type: null } : baseProfile;
```

Comment:
I guess this ternary check is in case the service profile happens to be nested inside of an real base profile and not just a nested profile (with type `null`)

Reply:
Added comment here